extensions/amp-base-carousel/1.0/base-carousel.js

```diff
@@ -95,9 +97,18 @@ function BaseCarouselWithRef(
   const setCurrentSlide =
     carouselContext.setCurrentSlide ?? setCurrentSlideState;
   const {slides, setSlides} = carouselContext;
+
   const scrollRef = useRef(null);
   const containRef = useRef(null);
   const contentRef = useRef(null);
+
+  const [axis, setAxis] = useState(
```

Comment:
Is state/effect needed here? Seems like we can derive all necessary state from the `orientation` prop w/o an extra re-rendering step. No?

Reply:
Good catch, removed.